paramiko/ecdsakey.py

```diff
@@ -283,12 +283,22 @@ def _from_private_key(self, file_obj, password):
         self._decode_key(data)
 
     def _decode_key(self, data):
-        try:
-            key = serialization.load_der_private_key(
-                data, password=None, backend=default_backend()
-            )
-        except (ValueError, AssertionError) as e:
-            raise SSHException(str(e))
+        pkformat, data = data
+        if pkformat == self.PRIVATE_KEY_FORMAT_ORIGINAL:
+            try:
+                key = serialization.load_der_private_key(
+                    data, password=None, backend=default_backend()
+                )
+            except (ValueError, AssertionError) as e:
+                raise SSHException(str(e))
+        elif pkformat == self.PRIVATE_KEY_FORMAT_OPENSSH:
+            curve, verkey, sigkey = self._uint32_cstruct_unpack(data, "sss")
+            try:
+                key = ec.derive_private_key(sigkey, curve, default_backend())
```

Comment:
this doesn't work: `sigkey` needs to be an integer, and `curve` needs to be translated into one of the Curve objects in `_ECDSA_CURVES`

noticed while working on my port and cleanup of this. here's a test to show the errors:
https://github.com/ploxiln/paramiko-ng/pull/13/commits/ab0389154ba4c083504fc97f3a973d09e08d5568
the fix is mixed into my large refactor, since I figured this out towards the end:
https://github.com/ploxiln/paramiko-ng/pull/13/commits/2a1b69e074897e0e08c6b3e2048b136f9a5a3fdd